src/eth.rs

```diff
@@ -482,6 +486,7 @@ impl Device {
 
         self.rx.init(eth_dma);
         self.tx.init(eth_dma);
+        cortex_m::asm::dsb();
```

Comment:
That barrier should be 10 lines further down before the DMAC[RT]X_CR enables to indicate the deadline.